File: bdep.js


let data = msg.payload['root']['station'][0]['etd'];
let times = "";
for(let i=0;i<=data.length;i++){
    let item = data[i];
    let destination = item['destination'];
    let trains = [];
    let est = item['estimate']
    for (let e=0;e<=est.length;i++){
        let eitem = est[e];
        trains.push(eitem['minutes'] + 'minutes');
    }
    times.push(destination + trains.join(', '));
}
return times;

File: bdep.py
import requests
import json

station = 'POWL'
r = requests.get('http://api.bart.gov/api/etd.aspx?cmd=etd&orig='+station+'&key=MW9S-E7SL-26DU-VV8V&json=y')
j = json.loads(r.text)

print('Trains from '+station+' station:')
for item in j['root']['station'][0]['etd']:
    destination = item['destination']
    trains = []
    est = item['estimate']
    for e in est:
        if e['minutes'] == 'Leaving':
            trains.append('Leaving now')
        else:
            trains.append(e['minutes'] + ' minutes')

    print(destination + ' : ' + ', '.join(trains))

